content/tutorials/Straightforward-Video-calls-with-WebRTC-and-Ably.textile

```diff
@@ -0,0 +1,408 @@
+# @_onConnect@: this method sets the class property @isConnected@ to true. This is an indicator that the peers have been connected.
+# @_onClose@: this method deletes the current connection instance from an object called @connections@, which we will define later on in the @ably-videocall.js@ file.
+# @_onStream@: this method is called when the video/audio stream from the other peer has been received.
+# @_onError@: should an error occur in our connection, this method will be called. Currently, we just log out the details of the error.
+
+h2.
+  default: Step 5 - Displaying online users
+
+Before we can make calls via WebRTC, we need to verify that the other peer we want to connect to is online, as WebRTC would not connect with offline peers.
+
+Create a file named @ably-videocall.js@ and add:
+
+```[javascript]
+    var membersList = []
+    var connections = {}
+    var currentCall
+    var localStream
+    var constraints = { video: true, audio: true }
+    var apiKey = 'XXX_API_KEY'
+    var clientId = 'client-' + Math.random().toString(36).substr(2, 16)
+    var realtime = new Ably.Realtime({ key: apiKey, clientId: clientId })
+    var AblyRealtime = realtime.channels.get('ChatChannel')
+    
+    AblyRealtime.presence.subscribe('enter', function(member) {
+        AblyRealtime.presence.get((err, members) => {
+            membersList = members
+            renderMembers()
+        })
+    })
+    AblyRealtime.presence.subscribe('leave', member => {
+        AblyRealtime.presence.get((err, members) => {
+            membersList = members
+            renderMembers()
+        })
+    })
+    AblyRealtime.presence.enter()
+    
+    function renderMembers() {
+        var list = document.getElementById('memberList')
+        var online = document.getElementById('online')
+        online.innerHTML = 'Users online (' + (membersList.length === 0 ? 0 : membersList.length - 1) + ')'
+        var html = ''
+        if (membersList.length === 1) {
+            html += '<li> No member online </li>'
+            list.innerHTML = html
+            return
+        }
+        for (var index = 0; index < membersList.length; index++) {
+            var element = membersList[index]
+            if (element.clientId !== clientId) {
+                html += '<li><small>' + element.clientId + ' <button class="btn btn-xs btn-success" onclick=call("' + element.clientId + '")>call now</button> </small></li>'
+            }
+        }
+        list.innerHTML = html
+    }
+```
+"See this step in Github":https://github.com/samuelayo/tutorials/commit/16dfd8527ccf01abeb12341842b7ca9b60f36e24
+
+**What is going on in the code block above?**
+
+In the code block above please note that you need to set the value of @apiKey@ to your Ably API key, we also defined a couple of variables, these are explained below:
```

Comment:
Replace with 'Before we move forward, please ensure that you have set the value of @apikey@ to your private Ably API key. The other variables are explained below:'